From 5a8bedeb595d1bf1d53e5cbd14816774cf4f41c0 Mon Sep 17 00:00:00 2001
From: Nikhil Badola <nikhil.badola@freescale.com>
Date: Tue, 9 Dec 2014 19:00:32 +0530
Subject: [PATCH 096/124] drivers: usb: fsl: Turn on cache snooping hardware in
 usb gadget

Turn on cache snooping hardware in usb gadget for PPC 64-bit Kernel

Signed-off-by: Nikhil Badola <nikhil.badola@freescale.com>
Change-Id: Iaa4049a6bde66038a99f8d716a983c7c41ae4594
Reviewed-on: http://git.am.freescale.net:8181/25529
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Suresh Gupta <suresh.gupta@freescale.com>
Reviewed-by: Matthew Weigel <Matthew.Weigel@freescale.com>
---
 drivers/usb/gadget/udc/fsl_udc_core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/fsl_udc_core.c b/drivers/usb/gadget/udc/fsl_udc_core.c
index a2ae4f7..3194aff 100644
--- a/drivers/usb/gadget/udc/fsl_udc_core.c
+++ b/drivers/usb/gadget/udc/fsl_udc_core.c
@@ -338,7 +338,8 @@ static int dr_controller_setup(struct fsl_udc *udc)
 	}
 #endif
 
-#if defined(CONFIG_PPC32) && !defined(CONFIG_NOT_COHERENT_CACHE)
+#if (defined(CONFIG_PPC32) || defined(CONFIG_PPC64)) && \
+	!defined(CONFIG_NOT_COHERENT_CACHE)
 	/* Turn on cache snooping hardware, since some PowerPC platforms
 	 * wholly rely on hardware to deal with cache coherent. */
 
-- 
2.1.0.27.g96db324

